### PR TITLE
Update DG - Tanishq

### PR DESCRIPTION
**Pull request by Tanishq4331**
Updated the use cases, added the SortOrder implementation

The following diagrams are included:

image

image

image

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -461,34 +565,94 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 4b. A task with the edited details already exists in the task list.
 
-    * 1a1. SalesNote shows an error message.
+    * 2b1. SalesNote shows an error message.
+
+      Use case resumes from step 3.
+
+
+#### Use case: Edit a client
```
Comment (ngchisern) [inline docs/DeveloperGuide.md:609]: Is it worth mentioning the case where the client's name is modified?

**Review by ngchisern — CHANGES_REQUESTED** [review]
Looks good. Added some suggestions below.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -273,15 +273,70 @@ The sequence diagram below shows the interaction within the Logic component for
...
+
+Orders are sorted using the `SortDescriptor`, which implements the `Comparator<Order>` interface.
+Its `compare` method uses the `SortField` and `SortOrdering` to compare orders based on the user specified arrangement.
+
+#### Usage
+
+The Sequence Diagram below illustrates the interactions within the `Logic` component for the `parseCommand("sortorder f/a o/asc")` API call. 
+Details about tokenizing the user input to retrieve the field and ordering have been omitted.
+
+![SortOrdersParserSequenceDiagram](images/SortOrdersParserSequenceDiagram.png)
```
[the remark this answers is not in the document]
Comment (Tanishq4331) [reply docs/DeveloperGuide.md:307]: Nice spot! I've updated it to
image

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Tanishq4331) [reply docs/DeveloperGuide.md:609]: Thanks, I've now included the case

**Review by ngchisern — APPROVED** [review]
LGTM!

**Pull request merged**